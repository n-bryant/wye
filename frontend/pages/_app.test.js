import {
  ACTIONS,
  CONTENT_OPTIONS,
  INITIAL_STATE,
  reducer,
  WyeWithState,
  WyeApp
} from "./_app";

describe("WyeWithState", () => {
  it("should render successfully", () => {
    const props = {
      Component: () => <div></div>,
      apollo: jest.fn(),
      pageProps: {}
    };

    const wrapper = shallow(<WyeWithState {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("WyeApp", () => {
  it("should render successfully", () => {
    const wrapper = shallow(<WyeApp />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("reducer", () => {
  it("should throw by default", () => {
    expect(reducer).toThrow();
  });

  it("should update the users state value if the action type is to set users", () => {
    const users = ["foo"];
    expect(
      reducer(INITIAL_STATE, { type: ACTIONS.SET_USERS, value: users })
    ).toMatchObject({
      content: INITIAL_STATE.content,
      users
    });
  });

  it("should update the content state value if the action type is to set content", () => {
    expect(
      reducer(INITIAL_STATE, {
        type: ACTIONS.SET_CONTENT,
        value: CONTENT_OPTIONS.RECOMMENDATIONS
      })
    ).toMatchObject({
      content: CONTENT_OPTIONS.RECOMMENDATIONS,
      users: INITIAL_STATE.users
    });
  });
});
