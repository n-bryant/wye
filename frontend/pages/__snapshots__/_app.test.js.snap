// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WyeApp should render successfully 1`] = `<WyeWithState />`;

exports[`WyeWithState should render successfully 1`] = `
<Fragment>
  <WithStyles(CssBaseline) />
  <NextNProgress
    color="#fff"
    height={3}
    startPosition={0.3}
    stopDelayMs={200}
  />
  <ApolloProvider
    client={[MockFunction]}
  >
    <ContextProvider
      value={
        Object {
          "dispatch": [Function],
          "state": Object {
            "content": "WELCOME",
            "users": Array [],
          },
        }
      }
    >
      <WithStyles(Page)>
        <Component />
      </WithStyles(Page)>
    </ContextProvider>
  </ApolloProvider>
</Fragment>
`;
